### PR TITLE
Warnint 64to32 6186 v23.1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6186

Describe changes:
- fix `-Wshorten-64-to-32` warnings for some files : output, stream

Some commits of #9840
#11580 next batch

Still to do afterwards :
- detect
- util

Victor, I let you review the stream stuff ;-)

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 22323

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/output-json-netflow.c [hunk 1]
```diff
@@ -193,7 +193,9 @@ static void NetFlowLogEveToServer(JsonBuilder *js, Flow *f)
     jb_set_string(js, "start", timebuf1);
     jb_set_string(js, "end", timebuf2);
 
-    int32_t age = SCTIME_SECS(f->lastts) - SCTIME_SECS(f->startts);
+    DEBUG_VALIDATE_BUG_ON((int64_t)(SCTIME_SECS(f->lastts) - SCTIME_SECS(f->startts)) < INT32_MIN ||
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-netflow.c:196]: Oh yes, we can simply have age as u64...

File: src/output-json-http.c [hunk 2]
```diff
@@ -299,14 +300,16 @@ static void EveHttpLogJSONExtended(JsonBuilder *js, htp_tx_t *tx)
         char status_string[status_size];
         BytesToStringBuffer(bstr_ptr(tx->response_status), bstr_len(tx->response_status),
                 status_string, status_size);
-        unsigned int val = strtoul(status_string, NULL, 10);
-        jb_set_uint(js, "status", val);
+        uint32_t val;
+        if (ByteExtractStringUint32(&val, 10, 0, status_string) == 0) {
+            jb_set_uint(js, "status", val);
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-http.c:305]: response_status_number is used in the regular case when it is > 0

Trying to drop this whole `else` case

File: src/output-json.c [hunk 3]
```diff
@@ -931,7 +921,8 @@ int OutputJSONMemBufferCallback(const char *str, size_t size, void *data)
         MemBufferExpand(memb, wrapper->expand_by);
     }
 
-    MemBufferWriteRaw((*memb), (const uint8_t *)str, size);
+    DEBUG_VALIDATE_BUG_ON(size > UINT32_MAX);
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json.c:924]: Nope, because libjannson `json_dump_callback` wants `size_t` :-/

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11705

**Pull request closed without merging**